src/mqt/predictor/utils.py

```diff
@@ -153,12 +40,37 @@ def timeout_handler(signum, frame):  # Custom signal handler
     return res
 
 
-def calc_eval_score_for_qc(qc_path: str, device: str):
-    try:
-        qc = QuantumCircuit.from_qasm_file(qc_path)
-    except Exception as e:
-        print("Fail in calc_eval_score_for_qc: ", e)
-        return get_width_penalty()
+def reward_crit_depth(qc):
+    (
+        program_communication,
+        critical_depth,
+        entanglement_ratio,
+        parallelism,
+        liveness,
+    ) = calc_supermarq_features(qc)
+    return 1 - critical_depth
+
+
+def reward_parallelism(qc):
+    (
+        program_communication,
+        critical_depth,
+        entanglement_ratio,
+        parallelism,
+        liveness,
+    ) = calc_supermarq_features(qc)
+    return critical_depth
+
+
+def reward_expected_fidelity(qc_or_path: str, device: str):
```

Comment:
since all of these functions have the same prefix it could potentially make sense to move them into a `reward.py` file/submodule and drop the prefix here. This creates a separation of concern and encapsulates functionality that belongs together instead of throwing a bunch of unrelated functions into a single `helper` file.

Reply:
Done.